### ISSUE TITLE
Better warning message for negative scopes and enum elements prefixed with "not_"

### ISSUE THREAD

**Issue by brenogazzola** [rails/rails#36272]
The negative scopes feature causes a warning message in models that declare an enum where one of their members is the negative of another, such as:

```
class Rsvp < ApplicationRecord
  enum status: [:going, :maybe, :not_going]
end
```

The enum generates a scope `:not_going` that returns a different result from the new negative scope for `:going`. Right now the enum scope is the one overwriting so it won't break existing code, but Rails complains about it:

```
WARN -- : Creating scope :not_going. Overwriting existing method Rsvp.not_going.
```

Suppressing the warning is not a good idea, since future changes might cause the negative scope to overwrite the enum scope. Also, if a developer decides to rename :not_going he might introduce bugs because pieces of code that relied on that scope will continue working (and maybe not failing certain tests), but returning a different result.

I think a more specific warning, about one enum member being the negative of another would help developers migrating from 5.2 to 6.0 stable - who might not be aware of this feature - understand what is causing the warning and that it's not a change they did to their code, but a new feature in Rails. Something like:

```
WARN --: An enum element in Rsvp uses the prefix 'not_'. This will cause a conflict with auto generated negative scopes.
```

**Comment by itsWill** [comment]
I think this is reasonable request, 

We can write a failing test case for this in `active_record/enum_test.rb` like: 

``` ruby
  test "enums with a negative condition log a warning" do
    original, stdout  = $stdout, StringIO.new
    expected_message = "An enum element in Rsvp uses the prefix 'not_'."\
      "This will cause a conflict with auto generated negative scopes."

    klass = Class.new(ActiveRecord::Base) do
      enum status: [:sent, :not_sent]
    end

    assert_match expected_message, $stout

  ensure
    $stdout = original
  end
```

And we can likely add the check for this in the `enum` method here: https://github.com/rails/rails/blob/master/activerecord/lib/active_record/enum.rb#L150

@brenogazzola would you like to take a stab at submitting a patch, and seeing if it's accepted? 

I wouldn't remove the error message that's logged afterwards ("WARN -- : Creating scope :not_going. Overwriting existing method Rsvp.not_going.") since it tells us exactly what is being overwritten, and that seems useful.

**Comment by brenogazzola** [comment]
Sure. That's going to be my first time doing this though, so I might need to try a few times until I can get it right. I believe the "[Contributing to Ruby on Rails](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html)" section in the documentation has everything I need, right?

**Comment by itsWill** [comment]
Should have everything to get you started :+1:

**Comment by nachoal** [comment]
Any updates on this deprecation @brenogazzola ? Would love to help and I think that the new message would help understand the issue and prepare for it.

**Comment by brenogazzola** [comment]
@nachoal It's already been fixed in #36380
